src/cryptonote_basic/cryptonote_basic_impl.h

```diff
@@ -74,16 +74,37 @@ struct address_parse_info {
     KV_MAP_SERIALIZABLE
 };
 
+// Strongly-typed money amount used to interface with the SQL DB when inserting
+// money into the DB. Amounts stored in the DB use a higher precision by a
+// factor of `BATCH_REWARD_FACTOR`.
+struct sql_db_money {
+
+    // Construct a money value from an atomic $COIN amount.
+    static sql_db_money coin_amount(uint64_t amount) { return {.amount = amount * BATCH_REWARD_FACTOR}; }
+
+    // Construct a money value from an atomic $COIN amount denoted in DB units. DB units have more
+    // precision to minimise integer division errors.
+    static sql_db_money db_amount(uint64_t amount)   { return {.amount = amount}; }
+
+    uint64_t to_coin() const { return amount / BATCH_REWARD_FACTOR; }
+
+    uint64_t to_db() const { return amount; }
+
+    bool operator==(const sql_db_money& rhs) const { return amount == rhs.amount; }
+
+    uint64_t amount{0};
+};
+
 struct batch_sn_payment {
     cryptonote::address_parse_info address_info{};
     eth::address eth_address{};
-    uint64_t amount;
+    sql_db_money amount;
```

Comment:
Perhaps we could have a:
```C++
uint64_t coin_amount() const { return amount.to_coin(); }
```
to slightly simplify all the later `b.amount.to_coin()` calls to `b.coin_amount()`

Reply:
Done